python/revng/internal/daemon/graphql.py

```diff
@@ -33,6 +33,7 @@ class Invalidation:
 
 
 executor = ThreadPoolExecutor(1)
+save_executor = ThreadPoolExecutor(1)
```

Comment:
1. Add a comment about the main role of `executor`, i.e., not hanging `uvicorn`. Stress that, in any case, even if we raise the number of threads the parallelism won't improve to the lock with the C API in most scenarios.
3. Option 1:
   1. keep `save_executor` (if you don't feel 100% confident about thread-safety of the involved classes.
   2. Add a note about the role of `save_executor`: `save_executor` enables us to run non-saving request in parallel when we're pushing stuff on S3.
   3. Add a note about the fact that `save_executor` could be merged in `executor`. In order to preserve the benefit above, we need to increase the number of threads in the pool.
4. Option 2: merge `save_executor` in `executor` and increase the number of threads.

Reply:
1. Done
2. Didn't implement
3. Implemented